src/main/java/com/microsoft/jenkins/artifactmanager/AzureArtifactManager.java

```diff
@@ -245,7 +245,7 @@ public String getUrl() {
     private static class UploadToBlobStorage extends MasterToSlaveFileCallable<Void> {
 
         public static final int MAX_QUEUE_SIZE_IN_NETTY = 500;
-        public static final int TIMEOUT = 30;
+        public static final int TIMEOUT = 600;
```

Comment:
have you confirmed this works in your case? should this be configurable to balance between not hanging jobs and larger uploads?

Reply:
yes,  it is working good for my usecase.
> should this be configurable to balance between not hanging jobs and larger uploads?
how about your opinion?
and it don't think it will hanging jobs because the timeout is still there.
if larger uploads taking too much time, I believe users should consider compressing files into a zip format before uploading them.

Reply:
lets try it

Reply:
sure, thank you.